### PR TITLE
Reactify

### PR DESCRIPTION
**Pull request by garrensmith**

### CODE REVIEW

**Comment by garrensmith** [comment]
This is pretty cool `+384 −2,112`

File: app/addons/fauxton/navigation/tests/componentsSpec.react.jsx [hunk 1]
```diff
@@ -19,26 +19,18 @@ import React from "react";
...
-    afterEach(function () {
-      ReactDOM.unmountComponentAtNode(container);
-    });
-
     it('dispatch TOGGLE_NAVBAR_MENU on click', function () {
-      TestUtils.Simulate.click(ReactDOM.findDOMNode(burgerEl));
+      const toggleMenu = sinon.spy();
+      const burgerEl = mount(<Views.Burger toggleMenu={toggleMenu} />);
+      burgerEl.simulate('click');
+      //TestUtils.Simulate.click(ReactDOM.findDOMNode(burgerEl));
```
Comment (robertkowalski) [inline app/addons/fauxton/navigation/tests/componentsSpec.react.jsx]: delete

File: app/core/routeObject.js [hunk 2]
```diff
@@ -14,325 +14,46 @@ import FauxtonAPI from "./base";
...
-  },
-
-  triggerBroadcast: function (eventName) {
-    var args = Array.prototype.slice.call(arguments);
-    this.trigger.apply(this, args);
-
-    args.splice(0, 1, eventName, this);
-    broadcaster.trigger.apply(broadcaster, args);
-  },
-
```
Comment (robertkowalski) [inline app/core/routeObject.js]: pure joy!

File: app/core/router.js [hunk 3]
```diff
@@ -48,32 +49,35 @@ export default Backbone.Router.extend({
...
+    const routeUrls = RouteObject.prototype.getRouteUrls();
 
-    _.each(routeUrls, function (route) {
-      this.route(route, route.toString(), function () {
-        var args = Array.prototype.slice.call(arguments),
-        roles = RouteObject.prototype.getRouteRoles(route),
-        authPromise = FauxtonAPI.auth.checkAccess(roles);
+    routeUrls.forEach(route => {
+      this.route(route, route.toString(), (...args) => {
+        //const args = Array.prototype.slice.call(arguments),
```
Comment (robertkowalski) [inline app/core/router.js]: delete comment

File: assets/index.underscore [hunk 4]
```diff
@@ -47,23 +47,11 @@
...
-    </div>
-  </div>
+  <div id="app"></div>
 
   <!-- Application source. -->
   <script src="<%= bundlejs %>" type="text/javascript"></script>
-  <% if (development) {%>
-    <script src="/templates.js" type="text/javascript"></script>
-  <%}%>
-
```
Comment (robertkowalski) [inline assets/index.underscore]: finally! :)

**Comment by robertkowalski** [comment]
+1

**Comment by robertkowalski** [comment]
merged!

**Pull request closed without merging**